test/visualize/AverageVisualizer.cpp

```diff
@@ -0,0 +1,37 @@
+#include "BoostUnitTest.h"
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+#include "visualize/QuadAverageHandler.h"
+#include "visualize/Visualizer.h"
+#include "visualize/XMLWriter.h"
+#include "TestStructure.h"
+
+namespace pt = boost::property_tree;
+using namespace NuTo;
+
+struct UnstructuredGridCheck
+{
+public:
+    static void Average(std::string filename)
+    {
+        pt::ptree tree;
+        pt::read_xml(filename, tree);
+        int numOfPoints = tree.get<int>("VTKFile.UnstructuredGrid.Piece.<xmlattr>.NumberOfPoints");
+        int numOfCells = tree.get<int>("VTKFile.UnstructuredGrid.Piece.<xmlattr>.NumberOfCells");
```

Comment:
Optional (since I have no better idea): This tests the `QuadAverageHandler`, the `Visualizer`, the `UnstucturedGrid` and the `XMLWriter`. As I said, I have no better idea. But I think there is a rule that somehow states that the code should be designed in a (unit)testable way. Maybe we can come up with something in the future.